nuxeo-runtime/nuxeo-runtime-kv/src/main/java/org/nuxeo/runtime/kv/KeyValueStoreProvider.java

```diff
@@ -33,6 +35,16 @@
     void initialize(KeyValueStoreDescriptor descriptor);
 
     /**
+     * Returns a {@link Set} of the keys contained in this Key/Value store provider.
+     * <p>
+     * This operation may be slow and should only be used for management or debug purposes.
+     *
+     * @return the set of keys
+     * @since 9.3
+     */
+    Set<String> keySet();
```

Comment:
What about returning a `Stream` instead ? All usage (except from tests) convert it to a `Stream`. Redis implementation uses `Stream` to construct this `Set`.

Reply:
Ok